src/js-same-type.ts

```diff
@@ -0,0 +1,16 @@
+import { CannotMerge } from './cannot-merge';
+import { type } from './type';
+
+export function jsSameType(
+  orig: any,
+  left: any,
+  right: any,
+): typeof orig | 'array' | 'null' | typeof CannotMerge {
```

Comment:
`typeof orig` here is `any`, which makes the entire return type the same as `any`.